Error during `brew test-bot --only-formulae` step on tap CI

Hi there,

I have a handful of taps I maintain, and I've been using the Homebrew-ordained CI solution created by `brew tap-new`.

In the past few days, I've had two unrelated formula on separate taps fail to build in the Linux portion of this CI run, with no clear way of fixing or debugging further, both during the `brew test-bot --only-formulae` step:

```
==> brew uninstall --force ticky/dreamcast/gditools3
Error: No such file or directory @ rb_file_s_link - (./gditools3--3.0.1.x86_64_linux.bottle.tar.gz, /home/runner/.cache/Homebrew/downloads/7b2a5e8cc188a648502bb6ef03004a5db2c384852c167ee93d9ca8a02a10a220--gditools3-3.0.1.x86_64_linux.bottle.tar.gz)
Error: Process completed with exit code 1.
```

[GitHub Actions run](https://github.com/ticky/homebrew-dreamcast/pull/9/checks?check_run_id=3011506550) (Will likely expire soon) • [Full run log archive](https://github.com/Homebrew/homebrew-test-bot/files/6798655/logs_42.zip)

```
==> brew uninstall --force ticky/playstation/paladin
Error: No such file or directory @ rb_file_s_link - (./paladin--1.3_1.x86_64_linux.bottle.2.tar.gz, /home/runner/.cache/Homebrew/downloads/a48a404125187f63ff6bb4915949b8ae21508e7ef8aa8b2090fa05ba519e3770--paladin-1.3_1.x86_64_linux.bottle.2.tar.gz)
Error: Process completed with exit code 1.
```

[GitHub Actions run](https://github.com/ticky/homebrew-playstation/pull/47/checks?check_run_id=3043187194) • [Full run log archive](https://github.com/Homebrew/homebrew-test-bot/files/6798656/logs_73.zip)

There's no obvious connection between these two formulae, and they appear to fail in the same way. There are also no other errors or warnings in the output that I can see which would suggest what's gone wrong here. In the case of the `gditools3` run, this occurred repeatedly under the same conditions.

What's the best way for me to help debug this issue and get things working?

Thanks!

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -52,15 +52,30 @@ jobs:
         if: matrix.os == 'ubuntu-latest'
         run: docker build -t brew .
 
-      - name: Run brew test-bot --only-formulae --test-default-formula
+      - run: brew test-bot --only-formulae-detect --test-default-formula
+        id: formulae-detect
+
+      - name: Run brew test-bot --only-formulae --junit --only-json-tab --skip-dependents --testing-formulae=${{ steps.formulae-detect.outputs.testing_formulae }}
+        id: brew-test-bot-formulae
         run: |
           if [ "$RUNNER_OS" = "Linux" ]; then
             docker run \
               -e GITHUB_ACTIONS -e GITHUB_BASE_REF -e GITHUB_REF -e GITHUB_REPOSITORY -e GITHUB_SHA \
               --rm brew \
-              brew test-bot --only-formulae --test-default-formula
+              brew test-bot --only-formulae --junit --only-json-tab --skip-dependents --testing-formulae=${{ steps.formulae-detect.outputs.testing_formulae }}
           else
-            brew test-bot --only-formulae --test-default-formula
+            brew test-bot --only-formulae --junit --only-json-tab --skip-dependents --testing-formulae=${{ steps.formulae-detect.outputs.testing_formulae }}
+          fi
+
+      - name: Run brew test-bot --only-formulae-dependents --junit --skipped-or-failed-formulae=${{ steps.brew-test-bot-formulae.outputs.skipped_or_failed_formulae }}
+        run: |
+          if [ "$RUNNER_OS" = "Linux" ]; then
+            docker run \
+              -e GITHUB_ACTIONS -e GITHUB_BASE_REF -e GITHUB_REF -e GITHUB_REPOSITORY -e GITHUB_SHA \
+              --rm brew \
+              brew test-bot --only-formulae-dependents --junit --skipped-or-failed-formulae=${{ steps.brew-test-bot-formulae.outputs.skipped_or_failed_formulae }}}
+          else
+            brew test-bot --only-formulae-dependents --junit --skipped-or-failed-formulae=${{ steps.brew-test-bot-formulae.outputs.skipped_or_failed_formulae }}
           fi
 
       - name: Output brew test-bot failures

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 coverage/
+vendor/
 bottle_output.txt
 brew-test-bot.xml
 steps_output.txt

--- a/lib/tests/formulae.rb
+++ b/lib/tests/formulae.rb
@@ -178,8 +178,7 @@ module Homebrew
         filename = bottle_json.dig(formula.full_name, "bottle", "tags").values.first["filename"]
 
         download_strategy = CurlDownloadStrategy.new("#{root_url}/#{filename}", formula.name, formula.version)
-
-        HOMEBREW_CACHE.mkpath
+        download_strategy.cached_location.parent.mkpath
         FileUtils.ln @bottle_filename, download_strategy.cached_location, force: true
         FileUtils.ln_s download_strategy.cached_location.relative_path_from(download_strategy.symlink_location),
                        download_strategy.symlink_location,

